regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
-      decimal_line_cols <- which(vapply(lines_to_plot, function(x) { is.numeric(x) && as.integer(x) != x }, FUN.VALUE = logical(1)))
-      DT::datatable(lines_to_plot, options = list(pageLength = 10), colnames = line_col_names, rownames = FALSE) %>%
-        DT::formatRound(columns = decimal_line_cols, digits=2)
-    })
-
-    # Creates data for the zones datatable
-    output$zones_data_table <- DT::renderDataTable({
-      # Reactive values that must trigger a tabel update
-      region$repopulate_region
-
-      if(is.null(to_plot$zones@data)){
-        return()
-      }
+  ## Hide/show panels on user-demand
+  shinyjs::onclick("toggle_panel", shinyjs::toggle(id = "input_panel", anim = FALSE))
+  shinyjs::onclick("toggle_imd_legend", shinyjs::toggle(id = "imd_legend", anim = FALSE))
 
-      zones_to_plot <- to_plot$zones@data[,unname(zone_col_names)]
-      decimal_zone_cols <- which(vapply(zones_to_plot, function(x) { is.numeric(x) && as.integer(x) != x }, FUN.VALUE = logical(1)))
-      DT::datatable(zones_to_plot, options = list(pageLength = 10),
-                    colnames = zone_col_names, rownames = FALSE) %>%
-        DT::formatRound(columns = decimal_zone_cols, digits=2)
-    })
+  ## Function to add a layers control for the routes, so that users can easily select quiet routes
+  # NB: currently this resets every time scenario changes or lines get redrawn because of map zoom - possibly to revisit
+  observe({
+    input$line_type
 
-    output$download_l_csv <- downloadHandler(
-      filename = function() { "lines.csv"  },
-      content = function(file) { write.csv(signif_sdf(to_plot$straight_line)@data[codebook_l$`Variable name`], file = file) }
-    )
+    if (input$line_type == 'routes') {
+      leafletProxy("map") %>% addLayersControl(
+        position = c("bottomright"),
+        overlayGroups = c("routes_fast", "routes_quieter"),
+        options = layersControlOptions(collapsed = T)
+      )
+    } else if (input$line_type != 'routes') {
+      leafletProxy("map") %>% showGroup(c("routes_fast", "routes_quieter")) %>% removeLayersControl()
+    }
+  })
 
-    output$download_z_csv <- downloadHandler(
-      filename = function() { "zones.csv"  },
-      content = function(file) { write.csv(signif_sdf(to_plot$zones)@data[codebook_z$`Variable name`], file = file) }
-    )
+  ## Read region_stats.html, if it exists, for the loaded region
+  output$region_stats <- renderUI({
+    input$purpose
+    input$geography
+    region$current
 
-    output$download_z_geojson <- downloadHandler(
-      filename = function() { "zones.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$zones[codebook_z$`Variable name`]), file = file) }
-    )
+    region_stats_file <-
+      file.path("../tabs/region_stats", input$purpose, input$geography, region$current,
+                "region_stats.html")
+    if (file.exists(region_stats_file))
```

Comment:
Sorry if I've missed it but where are the `region_stats.html` files are generated.

Reply:
Here: https://github.com/pctbike/pct-scripts/blob/master/07a_build_master.R

Reply:
Great, thanks.